lib/validators/doc-media-validator.ts

```diff
@@ -24,9 +30,26 @@ export class DocMediaValidator implements Validator {
         let valid = true;
         Object.values(this.definition.components).forEach((parsedComponent: ParsedComponentsDefinitionComponent) => {
 
-            if (this.countMediaDirectives(parsedComponent) > 1) {
+            let numMediaDirectives = this.countMediaDirectives(parsedComponent);
+            if (numMediaDirectives > 1) {
                 errorReporter(`A component can have only one "doc-media" directive in the HTML definition`);
                 valid = false;
+            } else {
+                let mediaProperties = Object.values(parsedComponent.properties).filter((property) => property.control.type === 'media-properties');
+                // Check whether the component has a media-properties control type
+                if (numMediaDirectives < 1 && mediaProperties.length > 0) {
+                    errorReporter(`Only components with a doc-media directive can have a "media-properties" control type`);
+                    valid = false;
+                }
+                if (numMediaDirectives === 1) {
+                    // Check whether the media-properties control type property is applied to the doc-media directive.
+                    Object.values(mediaProperties).forEach((mediaProperty) => {
+                        if (mediaProperty.directiveKey !== 'media') {
```

Comment:
Directive key can be any value. So you may check for exising of the media directive like
const directive = parsedComponent.directives[mediaProperty.directiveKey];
if (!directive || directive.type !== DirectiveType.media) {
  // property points to non exising directive or non doc-media directive
}